sdk/include/crt/crtdbg.h

```diff
@@ -210,11 +210,11 @@ extern "C" {
 }
 
   void *__cdecl operator new[](size_t _Size);
-  inline void *__cdecl operator new(size_t _Size,int,const char *,int) { return ::operator new(_Size); }
-  inline void *__cdecl operator new[](size_t _Size,int,const char *,int) { return ::operator new[](_Size); }
-  void __cdecl operator delete[](void *);
-  inline void __cdecl operator delete(void *_P,int,const char *,int) { ::operator delete(_P); }
-  inline void __cdecl operator delete[](void *_P,int,const char *,int) { ::operator delete[](_P); }
+  inline void *__cdecl operator new(size_t _Size,int,const char *,int) throw() { return ::operator new(_Size); }
```

Comment:
This is not correct. This operator and the one below will throw std::bad_alloc on failure.

Reply:
Hmm, the context was an attempt to address this:
```
In file included from e:/reactos/sdk/include/crt/new.h:11:0,
                 from e:/reactos/sdk/include/crt/comip.h:21,
                 from e:/reactos/sdk/include/crt/comdef.h:169,
                 from e:/reactos/base/applications/msconfig_new/xmldomparser.hpp:23,
                 from e:/reactos/base/applications/msconfig_new/toolspage.cpp:11,
                 from base/applications/msconfig_new/msconfig_new_unity.cpp:5:
e:/reactos/sdk/include/c++/new:41:43: error: declaration of 'void operator delete [](void*) throw ()' has a different exception specifier
In file included from e:/reactos/sdk/lib/atl/atlcore.h:35:0,
                 from e:/reactos/sdk/lib/atl/atlbase.h:23,
                 from e:/reactos/sdk/lib/atl/atlcoll.h:5,
                 from e:/reactos/base/applications/msconfig_new/srvpage.cpp:22,
                 from base/applications/msconfig_new/msconfig_new_unity.cpp:4:
e:/reactos/sdk/include/crt/crtdbg.h:215:16: error: from previous declaration 'void operator delete [](void*)'
```

Reply:
@ThFabba The context is, there is a mismatch between `sdk/include/c++/new` and `sdk/include/crt/crtdbg.h`, and this caused a problem for the unity build.

Basically, the `crtdbgh`. does not specify any `throw()` at all, and the `c++/new` specifies it on everything.

Reply:
Doing it to operator delete should be fine. But these two new operators reference ones that don't even exist in c++/new. Do the new ones really cause a similar warning? Can you show me that one?

Reply:
The warning was only the delete one, but I suggested to add it to the rest, because if you are going to fix something, better to fix the entire block in this case.
But because I was not sure at which function definitions to place it (I couldnt find a definitive answer on cppreference, I suggested to make it a PR)

Reply:
Updated.